File: backend/src/TaskMgmt.Domain/Entities/Comment.cs
using TaskMgmt.Domain.Common;

namespace TaskMgmt.Domain.Entities;

public class Comment : AuditableEntity
{
    public required Guid WorkTaskId { get; set; }
    public WorkTask? WorkTask { get; set; }

    public required string Content { get; set; }

    // Ánh xạ tới CreatedByUserId (kế thừa từ AuditableEntity) để hiển thị người viết bình luận.
    public User? Author { get; set; }

    public ICollection<CommentMention> Mentions { get; set; } = [];
}

File: mobile/taskmgmt_app/lib/features/comments/domain/entities/comment.dart
class CommentMention {
  const CommentMention({required this.userId, required this.fullName, required this.email});

  final String userId;
  final String fullName;
  final String email;
}

class Comment {
  const Comment({
    required this.id,
    required this.workTaskId,
    required this.content,
    required this.authorUserId,
    required this.authorFullName,
    required this.authorEmail,
    required this.authorHasAvatar,
    required this.createdAtUtc,
    required this.mentions,
  });

  final String id;
  final String workTaskId;
  final String content;
  final String? authorUserId;
  final String authorFullName;
  final String authorEmail;
  final bool authorHasAvatar;
  final DateTime createdAtUtc;
  final List<CommentMention> mentions;
}
